File: maths/addDigits.java
package maths;

public class addDigits{
    public static void main(String[] args) {
        SolutionAD solution = new SolutionAD();
        System.out.println(solution.addDigits(38));
    }
}

class SolutionAD {
    public int addDigits(int num) {
        
        while (num>9){
            
            int newnum =0;
            
            while(num>0){
                newnum+= num%10;
                num = num/10;
            }
            
            num = newnum;
        }
        
        return num;    
    }
}

File: maths/addDigits.py
# https://leetcode.com/problems/add-digits/

class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        
        while num > 9:
            temp = 0
            while num>0:
                temp+=num%10
                num = num//10
            num = temp
        return num

    # TC : O(1)
    # SC : O(1)
    def addDigits(self, num: int) -> int:

        if num == 0 :
            return 0
        elif num %  9 == 0:
            return 9
        else :
            return num % 9

         
        
print(Solution().addDigits(38))